catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/operation/impl/QueryResponseImpl.java

```diff
@@ -285,7 +290,11 @@ public Result take() {
     private Result handleTake() {
         Result result = null;
         try {
-            result = queue.take();
+            if (timeoutMillis <= 0) {
+                result = queue.take();
```

Comment:
Needed this here because the poll method will immediately return without waiting if given a timeout of 0